packages/dev/core/src/Meshes/abstractMesh.ts

```diff
@@ -2794,6 +2808,26 @@ export abstract class AbstractMesh extends TransformNode implements IDisposable,
     public getConnectedParticleSystems(): IParticleSystem[] {
         return this._scene.particleSystems.filter((particleSystem) => particleSystem.emitter === this);
     }
+
+    /**
+     * Compute a world space hotspot position
+     * @param hotSpotData point indices and barycentric
+     * @param res output world position
+     */
+    public getHotSpotToRef(hotSpotData: IHotSpotData, res: Vector3): void {
+        const options = { applySkeleton: true, applyMorph: true, updatePositionsArray: false };
+        const positions = this._getData(options, null, VertexBuffer.PositionKind);

```

Comment:
Do you see a way to not compute whole array but just the necessary positions @Popov72 ?

Reply:
Not in the current codebase, you would have to extract the code that applies skeleton/morph to a vertex from `applyMorph / applySkeleton` in `AbstractMesh` and apply it to your hotspot.